azure-iot-device/tests/common/pipeline/test_pipeline_stages_base.py

```diff
@@ -3170,67 +3239,55 @@ def stage(self, mocker, cls_type, init_kwargs, state, reconnect_timer, mock_time
-    def test_changes_state(self, stage, event):
-        stage.pipeline_root.connected = True
-        stage.handle_pipeline_event(event)
-        assert stage.state == pipeline_stages_base.ReconnectState.WAITING_TO_RECONNECT
-
-    @pytest.mark.it("If not previously connectred, does not change the state")
-    def test_does_not_change_state(self, stage, event):
-        stage.pipeline_root.connected = False
-        original_state = stage.state
-        stage.handle_pipeline_event(event)
-        assert stage.state == original_state
-
-    @pytest.mark.it("If previously connected, clears the previous reconnect timer if there was one")
-    def test_clears_reconnect_timer(self, stage, event):
-        old_timer = stage.reconnect_timer
-        stage.pipeline_root.connected = True
+    @pytest.mark.parametrize("state", [pipeline_stages_base.ReconnectState.LOGICALLY_CONNECTED])
+    @pytest.mark.it(
+        "If and only if connected and logically connected, changes the state to WAITING_TO_RECONNECT"
+    )
+    def test_changes_state(self, stage, event, state, currently_connected):
+        stage.pipeline_root.connected = currently_connected
         stage.handle_pipeline_event(event)
-        if old_timer:
-            assert old_timer.cancel.call_count == 1
-        assert stage.reconnect_timer != old_timer
+        if currently_connected and state == pipeline_stages_base.ReconnectState.LOGICALLY_CONNECTED:
+            assert stage.state == pipeline_stages_base.ReconnectState.WAITING_TO_RECONNECT
+        else:
+            assert stage.state == state
 
     @pytest.mark.it(
-        "If not previously connected, does not clears the previous reconnect timer if there was one"
+        "If and only if connected and logically connected, clears the previous reconnect timer if there was one"
     )
-    def test_does_not_clear_reconnect_timer(self, stage, event):
+    def test_clears_reconnect_timer(self, stage, event, state, currently_connected):
+        stage.pipeline_root.connected = currently_connected
         old_timer = stage.reconnect_timer
-        stage.pipeline_root.connected = False
         stage.handle_pipeline_event(event)
-        if old_timer:
-            assert old_timer.cancel.call_count == 0
-        assert stage.reconnect_timer == old_timer
 
-    @pytest.mark.it("If previously connected, sets a new reconnect timer")
-    def test_sets_new_reconnect_timer(self, stage, event, mock_timer):
-        stage.pipeline_root.connected = True
-        stage.handle_pipeline_event(event)
-        assert mock_timer.call_count == 1
-        assert stage.reconnect_timer == mock_timer.return_value
-        assert stage.reconnect_timer.start.call_count == 1
+        if currently_connected and state == pipeline_stages_base.ReconnectState.LOGICALLY_CONNECTED:
+            if old_timer:
+                assert old_timer.cancel.call_count == 1
+            assert stage.reconnect_timer != old_timer
```

Comment:
>!=  [](start = 41, length = 3)

are we trying to assert that these 2 objects are different ? then we should probably use `is not` for unequal identities.